design/backup_cancellation.md

```diff
@@ -0,0 +1,154 @@
+)
+```
+
+### Controller Changes
+
+#### Existing Controllers
+Modify `backup_controller.go`, `backup_operations_controller.go`, and `backup_finalizer_controller.go` to check for cancellation:
+```go
+// Early in each Reconcile method
+if backup.Spec.Cancel != nil && *backup.Spec.Cancel {
+    if backup.Status.Phase != BackupPhaseCancelling && backup.Status.Phase != BackupPhaseCancelled {
+        backup.Status.Phase = BackupPhaseCancelling
+        // Update backup and return
+        return ctrl.Result{}, c.Client.Patch(ctx, backup, client.MergeFrom(original))
+    }
+    return ctrl.Result{}, nil // Skip processing for cancelling/cancelled backups
+}
+```
+In addition, the `backup_operations_controller.go` will have a periodic check around backup progress updates, rather than running every time progress is updated to reduce API load.
+
+#### New Backup Cancellation Controller
+Create `backup_cancellation_controller.go`:
+```go
+type backupCancellationReconciler struct {
+    client.Client
+    logger                logrus.FieldLogger
+    itemOperationsMap     *itemoperationmap.BackupItemOperationsMap
+    newPluginManager      func(logger logrus.FieldLogger) clientmgmt.Manager
+    backupStoreGetter     persistence.ObjectBackupStoreGetter
+}
+```
+
+The controller will:
+1. Watch for backups in `BackupPhaseCancelling`
+2. Get operations from `itemOperationsMap.GetOperationsForBackup()`
+3. Call `bia.Cancel(operationID, backup)` on all in-progress BackupItemAction operations
+4. Find and cancel PodVolumeBackups by setting `pvb.Spec.Cancel = true`
+5. Wait for all cancellations to complete
+6. Set backup phase to `BackupPhaseCancelled`
+7. Update backup metadata in object storage
+
+### Cancellation Flow
+
+#### BackupItemAction Operations
+For operations with BackupItemAction v2 implementations (e.g., CSI PVC actions):
+1. Controller calls `bia.Cancel(operationID, backup)`
+2. CSI PVC action finds associated DataUpload and sets `du.Spec.Cancel = true`
+3. Node-agent DataUpload controller handles actual cancellation
+4. Operation marked as `OperationPhaseCanceled`
+
+#### PodVolumeBackup Operations
+For PodVolumeBackups (which lack BackupItemAction implementations):
+1. Controller directly finds PVBs by backup UID label
+2. Sets `pvb.Spec.Cancel = true` on in-progress PVBs
+3. Node-agent PodVolumeBackup controller handles actual cancellation
+
+
+## Alternatives Considered
+
+
+### Alternative 1: Deletion-Based Cancellation
+Using backup deletion as the cancellation mechanism instead of a cancel field.
+This was rejected because it doesn't allow users to preserve the backup object for inspection after cancellation, and deletion has different semantic meaning.
+
+### Alternative 2: Timeout-Based Automatic Cancellation
+Automatically cancelling backups after a configurable timeout.
+This was considered out of scope for the initial implementation as it addresses a different use case than user-initiated cancellation.
```

Comment:
Also, we already have timeout-based backup failure (both PVB and async BIAs have timeouts that trigger failure. That should be sufficient for timeout-based behavior.